apps/dashboard/src/components/workflow-editor/workflow-editor-provider.tsx

```diff
@@ -228,30 +257,37 @@ export const WorkflowEditorProvider = ({ children }: { children: ReactNode }) =>
   return (
     <WorkflowEditorContext.Provider value={value}>
       <AlertDialog open={blocker.state === 'blocked'}>
-        <AlertDialogContent>
+        <AlertDialogContent className="w-96">
           <AlertDialogHeader className="flex flex-row items-start gap-4">
-            <div className="bg-warning/10 rounded-lg p-3">
-              <RiAlertFill className="text-warning size-6" />
+            <div
+              className={`rounded-lg p-3 transition-all duration-300 ${
+                !isPending ? 'bg-success/10 scale-110' : 'bg-warning/10'
+              }`}
+            >
+              <div className="transition-opacity duration-300">
+                {!isPending ? (
+                  <CheckCircleIcon className="text-success animate-in fade-in size-6" />
+                ) : (
+                  <RiAlertFill className="text-warning animate-in fade-in size-6" />
+                )}
+              </div>
             </div>
             <div className="space-y-1">
-              <AlertDialogTitle>You might lose your progress</AlertDialogTitle>
-              <AlertDialogDescription>
-                This workflow has some unsaved changes. Save progress before you leave.
+              <div>
+                <AlertDialogTitle className="transition-all duration-300">
+                  {!isPending ? 'Changes saved!' : 'Saving changes'}
+                </AlertDialogTitle>
+              </div>
+              <AlertDialogDescription className="transition-all duration-300">
+                {!isPending ? 'Workflow has been saved successfully' : 'Please wait while we save your changes'}
               </AlertDialogDescription>
             </div>
+            {isPending && (
+              <button onClick={handleCancelNavigation} className="text-gray-500">
+                <RiCloseFill className="size-4" />
+              </button>
+            )}
```

Comment:
This should be an `<Dialog>` not `<AlertDialog>`

- Dialog - if user action is not that important -> has cancel button and cancel on background click by default
- AlertDialog - when user action is mandatory/important

In this case its more like a Dialog usecase but we have different styling for the `<Dialog>` and I didn't wanted to spent time unifying that in this PR - will refactor this in separate PR.

Reply:
Perfect. In the next iteration, I'd like us to examine the possibility of using a Sonner toast for this as well, so that we have one way to demonstrate feedback in the editor. It doesn't have to be centralized, and the no-click cloak can appear separately of the sonner message.